ansible/roles/prometheus-server/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+
+- name: ensure {{ prometheus_group }} group exists
+  become: yes
+  group:
+    name: "{{ prometheus_group }}"
+    system: yes
+
+- name: ensure {{ prometheus_user }} user exists
+  become: yes
+  user:
+    name: "{{ prometheus_user }}"
+    group: "{{ prometheus_user }}"
+    system: yes
+
+- name: check existing volume mounts
+  set_fact:
+    volume_mounted: "{{ ansible_mounts | selectattr('mount', 'match', prometheus_data_dir) | list | length > 0 }}"
+
+- name: configure volume mounts
+  include_role:
+    name: mounted-volume
+  when:
+    - not volume_mounted
+    - volume_mounts is defined
+    - volume_mounts != ""
+
+- name: prometheus directory structure
+  file:
+    state: "directory"
+    path: "{{ item }}"
+    owner: "{{ prometheus_user }}"
+    group: "{{ prometheus_group }}"
+  with_items:
+    - "{{ prometheus_root_dir }}"
+    - "{{ prometheus_dist_dir }}"
+    - "{{ prometheus_config_dir }}"
+    - "{{ prometheus_data_dir }}"
+
+- name: check if prometheus tarball exists locally
+  stat:
+    path: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+  register: prometheus_tar
+
+- name: download prometheus tarball when not exists
+  get_url:
+    url: "{{ prometheus_tar_gz_url }}"
+    dest: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+    validate_certs: "no"
```

Comment:
This comment may become redundant in light of others made regarding the retrieval of dependencies such as this from S3 in favour of external sources, but deliberately disabling SSL validation opens us up to exploitation as we would be dependent on an external source for this resource in addition to ignoring SSL certificates and are not performing any checksum verification either.

Reply:
Yep, good point. This task has now been replaced with the S3 download task, but fair point. Will keep that in mind for future.